transactions/src/main/java/org/jboss/as/txn/deployment/TransactionJndiBindingProcessor.java

```diff
@@ -57,26 +57,26 @@
     @Override
     public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        if(DeploymentTypeMarker.isType(DeploymentType.EAR, deploymentUnit)) {
+            return;
+        }
         final EEModuleDescription moduleDescription = deploymentUnit.getAttachment(Attachments.EE_MODULE_DESCRIPTION);
-
         if(moduleDescription == null) {
             return;
         }
-
         final ServiceTarget serviceTarget = phaseContext.getServiceTarget();
-        //if this is a war we need to bind to the modules comp namespace
-        if(DeploymentTypeMarker.isType(DeploymentType.WAR,deploymentUnit)) {
-            final ServiceName moduleContextServiceName = ContextNames.contextServiceNameOfModule(moduleDescription.getApplicationName(),moduleDescription.getModuleName());
-            bindServices(deploymentUnit, serviceTarget, moduleContextServiceName);
-        }
-
-        for(ComponentDescription component : moduleDescription.getComponentDescriptions()) {
-            if(component.getNamingMode() == ComponentNamingMode.CREATE) {
-                final ServiceName compContextServiceName = ContextNames.contextServiceNameOfComponent(moduleDescription.getApplicationName(),moduleDescription.getModuleName(),component.getComponentName());
-                bindServices(deploymentUnit, serviceTarget, compContextServiceName);
+        // bind to module
+        final ServiceName moduleContextServiceName = ContextNames.contextServiceNameOfModule(moduleDescription.getApplicationName(),moduleDescription.getModuleName());
+        bindServices(deploymentUnit, serviceTarget, moduleContextServiceName);
+        if (!DeploymentTypeMarker.isType(DeploymentType.WAR, deploymentUnit)) {
```

Comment:
This DUP didn't previously ignore non-WAR deployments.

Is this guard just an optimization because for a war component.getNamingMode() == ComponentNamingMode.CREATE will always be false?

Reply:
Aligning with the introduced abstract processor, honestly didn't dig if it's just an optimisation or a guard to possible misusage/extension where a war module may have a component that was lazily marked as having a comp naming context. This should probably should be further reworked to move logic present in way too many processors to the EE module and component description objects